File: Eulerpy/Euler3.py
#! /usr/bin/python


def is_prime(n):
	if(n % 2 == 0): return false
	for i in xrange(3, int(n**.5), 2):
		if n % i == 0:
			return False
	return True





number = 600851475143

for i in xrange(int(number ** .5)/2*2+1, 3, -2):
	if number % i == 0 and is_prime(i):
		print i
		break

File: Eulerpl/Euler3.pl
#!/usr/bin/perl

use warnings;
use strict;
use 5.010;
#600851475143
my $number = 600851475143;
my $largest = 0;
for(my $j = 1; $j < $number ** .5 ; $j += 2){
	
	if ($number % $j == 0 && &isPrime($j)==1) {
		$largest = $j;
}}
		print "$largest\n";



sub isPrime{
	my $param = shift;
	print "$param\n";

	for (my $i=2; $i< $param; $i ++){
		if ($param % $i == 0){
			return 0;  
		}
	}
	return 1;
}
